src/torsion_profile.py

```diff
@@ -0,0 +1,191 @@
+""" @package forcebalance.opt_geo_target Optimized Geometry fitting module.
+
+@author Yudong Qiu, Lee-Ping Wang
+@date 03/2019
+"""
+from __future__ import division
+import os
+import shutil
+import numpy as np
+import re
+import subprocess
+import itertools
+import json
+from copy import deepcopy
+from collections import OrderedDict, defaultdict
+from forcebalance.nifty import col, eqcgmx, flat, floatornan, fqcgmx, invert_svd, kb, printcool, printcool_dictionary, bohr2ang, warn_press_key
+from forcebalance.target import Target
+from forcebalance.molecule import Molecule
+from forcebalance.finite_difference import fdwrap, f1d2p, f12d3p, in_fd
+from forcebalance.output import getLogger
+from forcebalance.optimizer import Counter
+from forcebalance.opt_geo_target import periodic_diff
+logger = getLogger(__name__)
+
+class TorsionProfileTarget(Target):
+    """ Subclass of Target for fitting MM optimized geometries to QM optimized geometries. """
+    def __init__(self, options, tgt_opts, forcefield):
+        super(TorsionProfileTarget, self).__init__(options, tgt_opts, forcefield)
+
+        if not os.path.exists(os.path.join(self.tgtdir, 'metadata.json')):
+            raise RuntimeError('TorsionProfileTarget needs torsion drive metadata.json file')
+        with open(os.path.join(self.tgtdir, 'metadata.json')) as f:
+            self.metadata = json.load(f)
+            self.ndim = len(self.metadata['torsion_grid_ids'][0])
```

Comment:
I think it's more robust and easier to understand, if we load the dimension of torsion scan as `self.ndim = len(self.metadata['dihedrals'])`.

Reply:
Changes have been made in the latest commit.